minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/ConnectionSchema.java

```diff
@@ -60,30 +53,11 @@ public ConnectionSchema(Map map) {
         super(map, CONNECTIONS_KEY);
 
         sourceId = getOptionalKeyAsType(map, SOURCE_ID_KEY, String.class, CONNECTIONS_KEY, "");
```

Comment:
It is treated as required in the getValidationIssues implementation.  I needed a chance to set it when upconverting before the validation occurs.  I could change that to map manipulation before instantiating the ConnectionSchema but that felt worse.